### PR TITLE
#6 Command Callback Registration

### PR DESCRIPTION
**Pull request by misterinterrupt**
- cleanup
- Command Callback Registration

### CODE REVIEW

File: src/control_interface/src/command_register.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    command_map: CommandMap,
+}
+
+impl CommandRegister {
+    fn new() -> Self {
+        let mut map = CommandMap::new();
+        map.reserve(NUM_COMMANDS);
+        CommandRegister { command_map: map }
+    }
+    fn register_command(&mut self, command: *mut CString, registration: *mut c_void) {
```
Comment (misterinterrupt) [inline src/control_interface/src/command_register.rs:23]: perhaps CString should be 
```rust 
*const CStr
```

File: src/control_interface/src/command_register.rs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::ffi::CString;
+    #[test]
+    fn test_command_register() {
+        let mut command_register = CommandRegister::new();
+        let mut command = CString::new("test").unwrap();
```
Comment (ZavenArra) [inline src/control_interface/src/command_register.rs:36]: Consider replacing string key ("test" here) with a CommandIdentifier struct
{
  objectName,
  commandName
}

File: src/control_interface/src/command_register.rs [hunk 3]
```diff
@@ -0,0 +1,43 @@
+use core::ffi::c_void;
+use std::{collections::HashMap, ffi::CString};
+static NUM_COMMANDS: usize = 32;
+
+/// <div rustbindgen nocopy></div>
+/// <div rustbindgen opaque></div>
+#[repr(C)]
+pub struct RegisteredCommand {
```
Comment (misterinterrupt) [inline src/control_interface/src/command_register.rs:8]: register the command only

File: src/control_interface/src/command_register.rs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    object: *mut c_void,
+    action: *mut c_void,
+}
```
Comment (ZavenArra) [inline src/control_interface/src/command_register.rs:11]: Registered command may not need an 'object' but we might want to have a 'closure' object that contains the populated struct of the command payload as well as the 'action', i.e. function pointer.

File: src/control_interface/src/command_register.rs [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+    BoardMcuStop = 30,
+    BoardMcuSleep = 31,
+    BoardSignalExAdcHigh = 32,
+    BoardSignalExAdcLow = 33,
+    BoardSignal3v3BoostHigh = 34,
+    BoardSignal3v3BoostLow = 35,
+    Unknown = 36,
+}
+impl Command {
+    pub fn from_cstr(cstr: *const i8, len: usize) -> Self {
```
Comment (ZavenArra) [inline src/control_interface/src/command_register.rs:48]: from_attributes( object_string, action_string)

**Pull request merged**